dvc/exceptions.py

```diff
@@ -328,3 +329,12 @@ def __init__(self, target_infos):
 
 class CollectCacheError(DvcException):
     pass
+
+
+class NoOutputInExternalRepoError(DvcException):
+    def __init__(self, repo_url, path):
+        super(NoOutputInExternalRepoError, self).__init__(
+            "Output: '{}' not found in target repository: '{}'".format(
```

Comment:
Probably doesn't need `:` after `Output`, but otherwise this looks good to me. 🙂